source/AI.cpp

```diff
@@ -535,25 +526,44 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 	const int maxMinerCount = minables.empty() ? 0 : 9;
 	bool opportunisticEscorts = !Preferences::Has("Turrets focus fire");
 	bool fightersRetreat = Preferences::Has("Damaged fighters retreat");
+	bool shouldUpdateEscortState = !static_cast<bool>(Random::Int(20));
+	bool playerEscortStateUpdated = false;
 	for(const auto &it : ships)
```

Comment:
What's going on here? How is this relevant to fighters returning because they are low on energy?

Reply:
This is related to updating ship state and fleet state; battery power being a part.  There's a function called lower in AI UpdateEscortsState whose purpose is to calculate expensive Ship calculations and cache them in Ship instead of calculating every frame.

The benefit to doing it this way is fleet awareness for example refueling AI.  Which refuels the fleet in order of urgency for fuel.  This pre-caches other values like IsDisabled which gets called a lot.